### DISCUSSION TITLE
Any testers out there? [CheckBoxes]

### DISCUSSION THREAD

**Discussion by Dendraspis** [staxrip/staxrip#570] [Ideas]
Hey @44vince44 and @JJKylee ,
hello everybody,

this time I just need some feedback how my checkboxes work for you. Introducing themes made them somehow necessary.  🙄 
You will notice them on Main window as well as Jobs window.  Not perfect, but not as disturbing as the default ones I think.

Let my know what you think...

- [StaxRip v2.1.8.4D_Checkbox6.zip](https://github.com/staxrip/staxrip/files/6097528/StaxRip.v2.1.8.4D_Checkbox6.zip)
  - Code restored  🙄 
  - Updated to latest commit (23a123ddac2fcd38b16ca2a7ffb3e792f4978cb5)
- [StaxRip v2.1.8.4D_Checkbox5.zip](https://github.com/staxrip/staxrip/files/6097230/StaxRip.v2.1.8.4D_Checkbox5.zip)
  - Improvements
  - Fix disappearing text on Default theme
- [StaxRip v2.1.8.4D_Checkbox4.zip](https://github.com/staxrip/staxrip/files/6097150/StaxRip.v2.1.8.4D_Checkbox4.zip)
  - Improvements
  - Checkmark doesn't use theme color anymore
- [StaxRip v2.1.8.4D_Checkbox3.zip](https://github.com/staxrip/staxrip/files/6097094/StaxRip.v2.1.8.4D_Checkbox3.zip)
  - Improvements
  - Checkmark uses theme color
- [StaxRip v2.1.8.4D_Checkbox2.zip](https://github.com/staxrip/staxrip/files/6097039/StaxRip.v2.1.8.4D_Checkbox2.zip)
  - Normal Checkboxes added
- [StaxRip v2.1.8.4D_Checkbox1.zip](https://github.com/staxrip/staxrip/files/6096063/StaxRip.v2.1.8.4D_Checkbox1.zip)
  - Checkboxes in ListViews only

**Comment by ghost** [comment]
Someone has been drawing checkboxes 😁 
THEY ARE AMAZING 👍🏼 🚀 ✌🏼 🎉 

They look good and are relaxing to the eyes!

But maybe they must not be used in default mode....

**Comment by Dendraspis** [comment]
**V2 is out now**  🎉  🥳 

Is think this is way more eye friendly than before, especially at night. 😎 

### Before

<img width="466" src="https://user-images.githubusercontent.com/50659978/110239156-e5372700-7f45-11eb-873f-f3147d1b353b.PNG">

### After

<img width="464" src="https://user-images.githubusercontent.com/50659978/110239157-e5372700-7f45-11eb-9863-5022a2ddb39c.PNG">

**Comment by Dendraspis** [comment]
**v4 is out now** 🎉 🥳

...and I think we're almost done.  🤣

**Reply by ghost** [reply]
I'm on it in few minutes! (didn't have time to test v2 lol)

**Reply by ghost** [reply]
OK I have two comments, first one is purely aesthetic, the second is important:

## 1 the look of the new ticks is nice, it looks good at scale 1.15 - 1.80. At scale 1 they look slightly big, but they are still OK.

However, there is something  that is visible, that makes them look strange (even without zooming)

![SNAG-0002](https://user-images.githubusercontent.com/17828567/110241787-5fbe7180-7f5b-11eb-976b-1751015fd954.jpg)


## 2 IMPORTANT: in DEFAULT MODE, the text of selected options don't show (in any option window)

To reproduce it:
- Open staxrip, set THEME = DEFAULT
- Open x265 options, go to tab "Rate Control 3" first option is selected, doesn't show. Unselect it, it shows. Or go to any options windows, and tick, untick see what happens. ONLY IN DEFAULT THEME!

![SNAG-0004](https://user-images.githubusercontent.com/17828567/110241876-e1ae9a80-7f5b-11eb-9157-42936b82e5b5.jpg)

![SNAG-0003](https://user-images.githubusercontent.com/17828567/110241879-e410f480-7f5b-11eb-9362-2366c30d7351.jpg)

**Reply by Dendraspis** [reply]
The *not so nice* points are related to the thickness of the lines. 🤔 
Try v5, I improved them a tiny bit.

**Reply by ghost** [reply]
@Dendraspis The checkboxes look better indeed, the tick inside is less bulky. I like them like this ! And the disappearing text in default mode is fixed !! 👍🏼

**Comment by Dendraspis** [comment]
**v5 is out now**  🎉

**Reply by ghost** [reply]
I will prepare some encodes right now, let's see how everything works :-)

**Reply by ghost** [reply]
So far so good, everything ok, checkboxes are nice, flicker in the main window when closing code editor is acceptable.... everyhing ok. Highlights are nice .... 😁

**Reply by Dendraspis** [reply]
You won't believe it. I lost the whole progress of the last days.  😩
Made a little mistake (where I didn't know it was one) and forgot to backup.

Was there anything except CheckBoxes and the CodeEditor ?  😖

**Reply by ghost** [reply]
Indeed I can't believe it :-(
OMG you must be so upset :-(

- Code Editor -> the new layout with scrollbars (the join filters and stuff are in the commits, right?)
- Checkboxes, including the disappearing checkboxes when using scrollbar
- listview with select line always highlighted in the job window... and also the text appeared behind the area of the scrollbar.

I advise you to get some rest before doing anything...

**Reply by Dendraspis** [reply]
@44vince44 
Thanks, seems like everything works as intended now.  😊

**Reply by ghost** [reply]
I'm so relieved to hear that !!

**Reply by Dendraspis** [reply]
...and guess who is back.  😉 

I've uploaded v6 with the newest code base. If you find something, that worked before, but not now, please tell me.  😊

**Reply by ghost** [reply]
> ...and guess who is back. 😉

that is good news indeed 😃 

and the other good news is that v6 seems ok !!! just tested the checkboxes and the rest, all ok apparently!
👍🏼 
Well Done !!!! 🎉

**Reply by Dendraspis** [reply]
Damn good to read that.  😁 

Still can't believe the progress StaxRip made since the introduction of Themes.  With Themes, OH and CLH it feels totally different in my opinion, especially at night.

Make me angry I haven't made it earlier.  😂

**Reply by ghost** [reply]
Well that's a real achievement (as I said before 😁 ) now we can work and relax. I can't imagine myself ever using the default theme again 👍🏼 😁 🤓

**Reply by Dendraspis** [reply]
Indeed!
I use *Default* Theme only for testing and then I switch back. 😁
Same with OH ... won't use StaxRip without it anymore. 😁

To not bloat the issue:
Saw your mention on my Smartphone ... but Code Editor is finally working?  What was the incident?  🤔

**Reply by ghost** [reply]
I was afraid something was broken (the incident = what happened this afternoon, when you deleted something you shouldn't have)

So i made a test, but forgot to reset menu. the join feature wasn't working. TILL I MADE THE MENU RESET (almost forgot that one 😁 )

So all is ok! all works as it should be 👍🏼 

> Same with OH ... won't use StaxRip without it anymore. 😁

Obviously, they (OH, CLH) 're always turned ON here 😁

**Reply by Dendraspis** [reply]
Great!  So everything is working and is going to be included in the next BETA.  ☀️

**Reply by Dendraspis** [reply]
@44vince44 
Are you ZKSVince ?  🤔
Or is there another Vince who prefers AVS?!  🤗

**Reply by ghost** [reply]
@Dendraspis It's me alright 🤣 
I was teaching a french guy how to use staxrip ;-) with x264 :-(

**Reply by Dendraspis** [reply]
So you joined right after me.  😁
And I bet because you checked out the new Readme.md.  😊

**Reply by ghost** [reply]
Yes I was surprised there was a discord channel 🤣 
And I could test the desktop sharing, amazing it works even with my 4MBits connection !!! Discord is very powerful!
Have to go eat somthing, I spend several hours lol !!!!!

**Reply by Dendraspis** [reply]
Indeed ... with 140 members!!  I was shocked. 😂

Bon appetit.

**Reply by ghost** [reply]
Merci :-)
How come we did only learned about it today ?

**Reply by Dendraspis** [reply]
Because of the old text formatting I guess. This is why I changed it. I saw the discord first after I changed about 50%.  😂

**Reply by ghost** [reply]
Morning !!!

**Reply by stax76** [reply]
I don't know how to say it in English, Checkbox hat eine Stufe, ein Versatz:

![Untitled](https://user-images.githubusercontent.com/11410540/110891681-cd1f2900-82f2-11eb-97e9-2089f2473bbf.png)

Is it intended?

**Reply by Dendraspis** [reply]
I think I have to redesign the checkmark.

It's looking different everywhere. Mine look good and thicker, @44vince44 s' checkmarks are even thicker.  🤢

That's why I'm hoping to see screenshots when I see a new issue.  😁

But really depressing!

I would like to know why 4px can vary so much.

**Reply by Dendraspis** [reply]
@stax76 
How does the current version look like on your system?

**Reply by stax76** [reply]
It's much better and good enough for me. Thanks for tweaking it.

![ckeck2](https://user-images.githubusercontent.com/11410540/110979332-bfa28700-8364-11eb-8d39-e58cb70c771c.png)

![check3](https://user-images.githubusercontent.com/11410540/110979741-4a838180-8365-11eb-9a94-03710d77f9e3.png)

**Reply by Dendraspis** [reply]
Yeah, the problem is the small size, so I had to teak a lot to get something nice, but still good visible.
But this looks indeed pretty well.  😁

**Comment by JJKylee** [comment]
The new checkbox is the most welcome feature among other things that were introduced in my absence. 😄
I like it A LOT. 😉
You guys did a wonderful job! 👍👍👍👍👍
